File: Traffic Light (Python)/stage4/QueueThread.py
import os
import threading
import time


class QueueThread(threading.Thread):

    def __init__(self, interval: int, max_roads: int, name: str):
        super().__init__()
        self.setName(name)
        self.interval = interval
        self.roads = max_roads
        self.from_start = 0
        self.printing = False
        self.exit = False

    def run(self):
        while not self.exit:
            if self.printing:
                os.system('cls' if os.name == 'nt' else 'clear')
                print(f"! {self.from_start}s. have passed since system startup !")
                print(f"! Number of roads: {self.roads} !")
                print(f"! Interval: {self.interval} !")
                print(f"! Press \"Enter\" to open menu !")
            time.sleep(1)
            self.from_start += 1

File: Traffic Light/4/QueueThread.java
package traffic;

import java.io.IOException;

public class QueueThread extends Thread {
    int SIZE;
    int seconds;
    int fromStart = 0;
    boolean menu = true;
    boolean exit = false;

    QueueThread(int size, int interval){
        SIZE = size;
        seconds = interval;
    }

    @Override
    public void run() {

        while (!exit) {
            try {
                this.sleep(1000);
            } catch (InterruptedException e) {
                e.printStackTrace();
            }
            fromStart++;
            if(!menu) {
                try {
                    if (System.getProperty("os.name").contains("Windows"))
                        new ProcessBuilder("cmd", "/c", "cls").inheritIO().start().waitFor();
                    else
                        Runtime.getRuntime().exec("clear");
                } catch (IOException | InterruptedException e) {
                }
                System.out.println("! " + fromStart + "s. have passed since system startup !");
                System.out.println("! Number of roads: " + SIZE + " !");
                System.out.println("! Interval: " + seconds + " !\n");
                System.out.println("\nPress \"Enter\" to open menu");
            }
        }
    }
}
